tools/IceRpc.BuildTelemetry.Reporter/TelemetryExtensions.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) ZeroC, Inc.
+
+using System.Diagnostics;
+using System.Reflection;
+
+namespace IceRpc.BuildTelemetry.Reporter;
+
+public partial record struct Telemetry
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Telemetry" /> struct using the specified command-line arguments.
+    /// </summary>
+    /// <param name="args">The command-line arguments.</param>
+    public Telemetry(string[] args)
+    {
+        // Determine the IceRPC version using the assembly version
+        var assembly = Assembly.GetAssembly(typeof(Telemetry));
+        string version = assembly?.GetName()?.Version?.ToString() ?? "unknown";
+
+        // Parse the source
```

Comment:
I don't know what this `source` is, even after reading the code below.

We need a clearer name.

Reply:
Added a comment